compendium/LogService/include/cppmicroservices/logservice/Logger.hpp

```diff
@@ -0,0 +1,110 @@
+	    virtual void debug(const std::string& format, const std::string& arg1, const std::string& arg2) = 0;
+	    virtual void debug(std::string const& message, const std::exception_ptr ex) = 0;
+	    virtual void debug(std::string const& message, ServiceReferenceBase const& sr) = 0;
+	    virtual void debug(std::string const& message, ServiceReferenceBase const& sr, const std::exception_ptr ex) = 0;
+
+	    virtual void error(const std::string& message) = 0;
+            virtual void error(const std::string& format, const std::string& arg) = 0;
+            virtual void error(const std::string& format, const std::string& arg1, const std::string& arg2) = 0;
+	    virtual void error(std::string const& message, const std::exception_ptr ex) = 0;
+            virtual void error(std::string const& message, ServiceReferenceBase const& sr) = 0;
+            virtual void error(std::string const& message, ServiceReferenceBase const& sr, const std::exception_ptr ex) = 0;
+
+	    virtual void info(const std::string& message) = 0;
+            virtual void info(const std::string& format, const std::string& arg) = 0;
+            virtual void info(const std::string& format, const std::string& arg1, const std::string& arg2) = 0;
+	    virtual void info(std::string const& message, const std::exception_ptr ex) = 0;
+            virtual void info(std::string const& message, ServiceReferenceBase const& sr) = 0;
+            virtual void info(std::string const& message, ServiceReferenceBase const& sr, const std::exception_ptr ex) = 0;
+
+	    virtual void trace(const std::string& message) = 0;
+            virtual void trace(const std::string& format, const std::string& arg) = 0;
+            virtual void trace(const std::string& format, const std::string& arg1, const std::string& arg2) = 0;
+	    virtual void trace(std::string const& message, const std::exception_ptr ex) = 0;
+            virtual void trace(std::string const& message, ServiceReferenceBase const& sr) = 0;
+            virtual void trace(std::string const& message, ServiceReferenceBase const& sr, const std::exception_ptr ex) = 0;
+
+	    virtual void warn(const std::string& message) = 0;
+            virtual void warn(const std::string& format, const std::string& arg) = 0;
+            virtual void warn(const std::string& format, const std::string& arg1, const std::string& arg2) = 0;
+	    virtual void warn(std::string const& message, const std::exception_ptr ex) = 0;
+            virtual void warn(std::string const& message, ServiceReferenceBase const& sr) = 0;
+            virtual void warn(std::string const& message, ServiceReferenceBase const& sr, const std::exception_ptr ex) = 0;
+
+	     /**
+             * Registers a sink to the logger for introspection of contents. This is not a publicly available
+             * function and should only be used for testing. This is NOT thread-safe.
+             */
+	    virtual void AddSink(spdlog::sink_ptr& sink) = 0;
```

Comment:
Typically, a logger interface would focus solely on logging actions, not on managing logger sinks. Does it violates SRP?

Reply:
This function registers a sink to the logger for introspection of contents. This is not a publicly available
 function and is only used for testing. 
We are using this function only for testing purpose.